Introduce Dashicon component

aduth opened the pull request:
This pull request seeks to add a new `<Dashicon />` component, sourced with revisions from the new Dashicons build process introduced at https://github.com/WordPress/dashicons/pull/169 . 

__Open Questions:__

- Relative paths will be a pain to work with for general-purpose components (`import Dashicon from '../../components/dashicon';`). We might want to look again at defining per-module root paths.

aduth committed:
Introduce Dashicon component

jasmussen commented:
Works great! 

`import Dashicon from '../components/dashicon';` and `<Dashicon icon="yes" />` seem to work exactly as intended. Ship it!

jasmussen committed:
Polish CSS a bit

- Add current color fill to SVGs everywhere
- Vertically center content in main toolbar

aduth commented on editor/header/mode-switcher/style.scss:
```diff
@@ -10,9 +10,9 @@
-	vertical-align: middle;
-	line-height: 20px;
 	outline: none;
+	display: flex;
```
Are we assuming IE11? 😄

jasmussen committed:
Fix travis errors.
